v2/cmd/controller/logging/logging.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package logging
+
+import (
+	"flag"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/klog/v2/textlogger"
+)
+
+type Config struct {
+	// verbose indicates the level of logging.
+	// Higher values indicate more logging.
+	verbose int
+
+	// useJSON indicates whether we should output logs in JSON format.
+	// Default is no
+	useJSON bool
+}
+
+// Create returns a new logger, ready for use.
+// This can be called multiple times if required.
+func Create(cfg *Config) logr.Logger {
+	if cfg != nil && cfg.useJSON {
+		log, err := createJSONLogger(cfg)
+		if err != nil {
+			log = createTextLogger(cfg)
+			log.Error(err, "failed to create JSON logger, falling back to text")
+		}
+
+		return log
+	}
+
+	return createTextLogger(cfg)
+}
+
+func createTextLogger(cfg *Config) logr.Logger {
+	opts := []textlogger.ConfigOption{}
+	if cfg != nil {
+		opts = append(opts, textlogger.Verbosity(cfg.verbose))
+	}
+
+	c := textlogger.NewConfig(opts...)
+	return textlogger.NewLogger(c)
+}
+
+func createJSONLogger(cfg *Config) (logr.Logger, error) {
+	level := zap.InfoLevel
+	if cfg != nil {
+		switch cfg.verbose {
+		case 0:
+			level = zap.ErrorLevel
+		case 1:
+			level = zap.WarnLevel
+		case 2:
+			level = zap.InfoLevel
+		case 3:
```

Comment:
\>= 3 can (I think?) be configured? Probably these should all be zap.DebugLevel, since users may ask to loglevel 10 (at least I think that's as high as controller-runtime goes)

Reply:
Changed.